src/plugins/vis_augmenter/server/routes/stats_helpers.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { get } from 'lodash';
+import {
+  SavedObjectsFindResponse,
+  SavedObjectsClientContract,
+} from '../../../../../src/core/server';
+import { AugmentVisSavedObjectAttributes } from '../../common';
+
+interface ObjectBreakdown {
+  origin_plugin: { [key: string]: number };
+  plugin_resource_type: { [key: string]: number };
+  plugin_resource_id: { [key: string]: number };
+  visualization_id: { [key: string]: number };
+}
+
+interface VisAugmenterStats {
+  total_objs: number;
+  obj_breakdown: ObjectBreakdown;
+}
+
+export const getAugmentVisSavedObjects = async (
+  savedObjectsClient: SavedObjectsClientContract,
+  perPage: number
+): Promise<SavedObjectsFindResponse<AugmentVisSavedObjectAttributes>> => {
+  const augmentVisSavedObjects: SavedObjectsFindResponse<AugmentVisSavedObjectAttributes> = await savedObjectsClient?.find(
+    {
+      type: 'augment-vis',
+      perPage,
+    }
+  );
+  // If there are more than perPage of objs, we need to make additional requests
+  if (augmentVisSavedObjects.total > perPage) {
+    const iterations = Math.ceil(augmentVisSavedObjects.total / perPage);
+    for (let i = 1; i < iterations; i++) {
+      const augmentVisSavedObjectsPage: SavedObjectsFindResponse<AugmentVisSavedObjectAttributes> = await savedObjectsClient?.find(
+        {
+          type: 'augment-vis',
+          perPage,
+          page: i + 1,
+        }
+      );
+      augmentVisSavedObjects.saved_objects = [
+        ...augmentVisSavedObjects.saved_objects,
+        ...augmentVisSavedObjectsPage.saved_objects,
+      ];
+    }
+  }
+  return augmentVisSavedObjects;
+};
+
+/**
+ * Given the _find response that contains all of the saved objects, iterate through them and
+ * increment counters for each unique value we are tracking
```

Comment:
We should add some description of what the output exactly means. Maybe providing an example output would help clear things more.

Reply:
So this is defined within the `resp` argument type - `SavedObjectsFindResponse`. Prefer to not add an example here which could become outdated. By looking up the defined type, it will show the fields.

Also, the saved object list is located under `saved_objects` field as seen on line 67